cdap-ui/app/cdap/components/KeyValuePairs/index.js

```diff
@@ -112,12 +112,6 @@ export default class KeyValuePairs extends Component {
     });
   }
   shouldComponentUpdate(nextProps) {
-    if (this.state.pairs.length !== nextProps.keyValues.pairs.length) {
```

Comment:
@ajainarayanan do you remember the context for this? I think this got added because there was an infinite loop.

Reply:
yeah talked to @tonybach. I think after react upgrade that got fixed. I think we should get rid of the `shouldComponentUpdate`. I don't think thats necessary.

Reply:
There is another blocker JIRA because of `shouldComponentUpdate` which was causing the keyvalue pairs to not update in UI. @tonybach can you provide the JIRA that you had filed?

Reply:
It's https://issues.cask.co/browse/CDAP-14332

Reply:
Confirmed that removing `shouldComponentUpdate` is not needed, and it fixes the JIRA I just linked as well. Can you guys try to see if there are any issues